### PR TITLE
Issue #7743: Update AbstractChecks to log DetailAST - MethodCount

### LINKED ISSUE #7743: Update AbstractChecks to log DetailAST - MethodCount
Child of https://github.com/checkstyle/checkstyle/issues/5777

This issue specifically focuses on MethodCount.

### PR DESCRIPTION
**Pull request by shashwat70**
Closes #7743: Update AbstractChecks to log DetailAST - MethodCount
Diff report: https://shashwat70.github.io/diff1/

### CODE REVIEW

File: src/it/resources/org/checkstyle/suppressionxpathfilter/methodcount/SuppressionXpathRegressionMethodCount3.java [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package org.checkstyle.suppressionxpathfilter.methodcount;
+
+class SuppressionXpathRegressionMethodCount3 { // warn
```
Comment (esilkensen) [inline src/it/resources/org/checkstyle/suppressionxpathfilter/methodcount/SuppressionXpathRegressionMethodCount3.java:3]: this looks like the same file as `SuppressionXpathRegressionMethodCount1` - could that one be reused instead?

Comment (shashwat70) [reply src/it/resources/org/checkstyle/suppressionxpathfilter/methodcount/SuppressionXpathRegressionMethodCount3.java:3]: Yes! You're right. I'll just fix it.

Comment (shashwat70) [reply src/it/resources/org/checkstyle/suppressionxpathfilter/methodcount/SuppressionXpathRegressionMethodCount3.java]: Done! @esilkensen

Comment (esilkensen) [reply src/it/resources/org/checkstyle/suppressionxpathfilter/methodcount/SuppressionXpathRegressionMethodCount3.java]: Cool, thanks. Code changes and diff report look good - now that the 3rd test input file is removed, the numbers on the 4th 5th could be decremented.

Alternatively, what do you think about a single test input file with public/private/protected/default methods, and separate test cases for each type of method - would that work?

Comment (shashwat70) [reply src/it/resources/org/checkstyle/suppressionxpathfilter/methodcount/SuppressionXpathRegressionMethodCount3.java]: Yes, that will work. But a look at https://github.com/checkstyle/checkstyle/issues/7987. I think it expects two or more. What do you say?

Comment (shashwat70) [reply src/it/resources/org/checkstyle/suppressionxpathfilter/methodcount/SuppressionXpathRegressionMethodCount3.java]: Or is it only for test methods and not input files?

Comment (esilkensen) [reply src/it/resources/org/checkstyle/suppressionxpathfilter/methodcount/SuppressionXpathRegressionMethodCount3.java]: I think Roman’s comment there refers to having two or more test methods, but if you want to rename the 4th and 5th input files to 3 and 4 respectively, I am good to approve PR. Maybe another reviewer has an opinion about combining them into one file - either way seems ok to me.

Comment (shashwat70) [reply src/it/resources/org/checkstyle/suppressionxpathfilter/methodcount/SuppressionXpathRegressionMethodCount3.java]: Done! Thanks for the review.

File: src/it/resources/org/checkstyle/suppressionxpathfilter/methodcount/SuppressionXpathRegressionMethodCount3.java [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package org.checkstyle.suppressionxpathfilter.methodcount;
+
+class SuppressionXpathRegressionMethodCount4 { // warn
```
Comment (esilkensen) [inline src/it/resources/org/checkstyle/suppressionxpathfilter/methodcount/SuppressionXpathRegressionMethodCount3.java:3]: rename class name to match file name

File: src/it/resources/org/checkstyle/suppressionxpathfilter/methodcount/SuppressionXpathRegressionMethodCount4.java [hunk 3]
```diff
@@ -0,0 +1,6 @@
+package org.checkstyle.suppressionxpathfilter.methodcount;
+
+class SuppressionXpathRegressionMethodCount5 { // warn
```
Comment (esilkensen) [inline src/it/resources/org/checkstyle/suppressionxpathfilter/methodcount/SuppressionXpathRegressionMethodCount4.java:3]: rename class name to match file name

File: src/it/resources/org/checkstyle/suppressionxpathfilter/methodcount/SuppressionXpathRegressionMethodCount3.java (hunk 2 above)
Comment (shashwat70) [reply src/it/resources/org/checkstyle/suppressionxpathfilter/methodcount/SuppressionXpathRegressionMethodCount3.java]: My bad. Done.

File: src/it/resources/org/checkstyle/suppressionxpathfilter/methodcount/SuppressionXpathRegressionMethodCount4.java (hunk 3 above)
Comment (shashwat70) [reply src/it/resources/org/checkstyle/suppressionxpathfilter/methodcount/SuppressionXpathRegressionMethodCount4.java]: Done.

**Pull request merged**